servicetalk-http-api/src/main/java/io/servicetalk/http/api/HttpServerBuilder.java

```diff
@@ -480,6 +480,13 @@ public final ServerContext listenBlockingStreamingAndAwait(
         return listenForService(adapterHolder.adaptor(), adapterHolder.serviceInvocationStrategy());
     }
 
+    /**
+     * Returns true if offloading is appropriate for the execution context.
+     *
+     * @return true if offloading is appropriate for the execution context.
+     */
+    protected abstract boolean shouldOffload();
```

Comment:
Can we defer adding this `protected` method? Consider starting with hardcoded check inside this class and we can make it `protected` on-demand.

Reply:
Originally the method was going to be specific to Netty implementation and this was the abstraction and I am still thinking that perhaps it should be. The `IoExecutor.IoThread` marker interface doesn't feel like it should be public.

Reply:
We do not have any known use-cases when users extend `HttpServerBuilder`. Therefore, proposing to defer expanding public API until we have a use-case.